Add AWS naming convention for IAM roles (#105)
giantswarm/fmt: aws/naming_resources.md

# Naming Resources

## IAM Resources

> AWS operator needs a set of roles in customer accounts to be able to operator against the AWS API. Most of the customers need bounded policy to be security complaint. Here we define a convention to help them to achieve that and a same time have a coherent way to name the roles. From now own all clusters roles we need to create should have the prefix `gs-`. Example:

Reviewer: Let's please wrap paragraphs by some amount of characters. 80 or 120 works well.

Author: done

OLD worker role: `5md7e-worker-EC2-K8S-Role`

> NEW worker role: `gs-5md7e-worker-EC2-K8S-Role`

Reviewer: Can we write it down schematically next to having examples? 

```
gs-[cluster id]-[what is this?]-EC2-K8S-Role
```

Reviewer: I would really like to see that scheme in there. Three people liked it apparently so let's get it in as is.

Author: But this is the schema, right?
https://github.com/giantswarm/fmt/pull/105/files#diff-1d117290cfad3e20761fe775d42fff1aR9-R11

Author: ok I made it more explicit, does it better?

# Naming Resources

## IAM Resources

> AWS operator needs a set of roles in customer accounts to be able to operator against the AWS API. Most of the customers need bounded policy to be security complaint. Here we define a convention to help them to achieve that and a same time have a coherent way to name the roles.

Reviewer 2: ```suggestion
AWS operator needs a set of roles in customer accounts to be able to act against the AWS API. Most of the customers need a bounded policy to be security complaint. Here we define a convention to help them to achieve this and at the same time have a coherent way to name the roles.
```

> AWS operator needs a set of roles in customer accounts to be able to operator against the AWS API. Most of the customers need bounded policy to be security complaint. Here we define a convention to help them to achieve that and a same time have a coherent way to name the roles.

Reviewer: The text is still not wrapped. Let me know when you need help with that.

Author: Hmm according to Visual code it is 
<img width="777" alt="Screen Shot 2020-04-08 at 16 38 05" src="https://user-images.githubusercontent.com/892157/78796972-5e856d00-79b7-11ea-8c1d-0a0ea06ffbad.png">

Author: done

> From now own all clusters roles we need to create should have this convention

Reviewer 2: ```suggestion
From now on all clusters roles we need to create should have this convention
```

- Common prefix `gs-`

> - Follow with `cluster id`

Author: ```suggestion
- Follow with `-cluster_id-`
```

> - Ending with `purpose`

Author: ```suggestion
- Ending with `-purpose`
```

Example:

> OLD worker role: `5md7e-worker-EC2-K8S-Role`

Author: ```suggestion
OLD worker role: `5md7e-master-EC2-K8S-Role`
```

> NEW worker role: `gs-5md7e-worker-EC2-K8S-Role`

Author: ```suggestion
NEW worker role: `gs-5md7e-master-ec2-k8s` (  gs        5md7e      master-ec2-k8s   )
                                            prefix   cluster_id      purpose
```
